cli/replace.php

```diff
@@ -0,0 +1,128 @@
+
+define('CLI_SCRIPT', true);
+
+require(__DIR__.'/../../../../config.php');
+require_once($CFG->libdir.'/clilib.php');
+require_once($CFG->libdir.'/adminlib.php');
+require_once($CFG->dirroot . '/lib/csvlib.class.php');
+$help =
+    "Replace strings using uploaded CSV file..
+
+Options:
+--input=FILE                  Required. Input CSV file produced by find.php in detail mode.
+-h, --help                    Print out this help.
+
+Example:
+\$ sudo -u www-data /usr/bin/php admin/tool/advancedreplace/cli/replace.php --input=/tmp/result.csv
+";
+
+list($options, $unrecognized) = cli_get_params(
+    [
+        'input'        => null,
+        'help'         => false,
+    ],
+    [
+        'h' => 'help',
+    ]
+);
+core_php_time_limit::raise();
+
+if ($unrecognized) {
+    $unrecognized = implode("\n  ", $unrecognized);
+    cli_error(get_string('cliunknowoption', 'admin', $unrecognized));
+}
+
+// Ensure that we have required parameters.
+if ($options['help'] || empty($options['input'])) {
+    echo $help;
+    exit(0);
+}
+
+try {
+    $file = validate_param($options['input'], PARAM_RAW);
+} catch (invalid_parameter_exception $e) {
+    cli_error(get_string('errorinvalidparam', 'tool_advancedreplace'));
+}
+
+if (!file_exists($file)) {
+    cli_error(get_string('errorfilenotfound', 'tool_advancedreplace'));
+}
+
+// Open the file for reading.
+$fp = fopen($file, 'r');
+$data = fread($fp, filesize($file));
+fclose($fp);
+
+// Load the CSV content.
+$iid = csv_import_reader::get_new_iid('tool_advancedreplace');
+$csvimport = new csv_import_reader($iid, 'tool_advancedreplace');
+$contentcount = $csvimport->load_csv_content($data, 'utf-8', 'comma');
+
+// Read the header and find the column indexes.
+$header = $csvimport->get_columns();
+if (empty($header)) {
+    cli_error(get_string('errorinvalidfile', 'tool_advancedreplace'));
+}
+
+$tableindex = array_search('table', $header);
+$columnindex = array_search('column', $header);
+$idindex = array_search('id', $header);
+$matchindex = array_search('match', $header);
+$replaceindex = array_search('replace', $header);
+
+if ($tableindex === false || $columnindex === false || $idindex === false || $matchindex === false || $replaceindex === false) {
+    cli_error(get_string('errorinvalidfile', 'tool_advancedreplace'));
+}
+
+// Read the data and replace the strings.
+$dataset = [];
+$csvimport->init();
+while ($record = $csvimport->next()) {
+    $dataset[] = $record;
+    $table = $record[$tableindex];
+    $columnname = $record[$columnindex];
+    $id = $record[$idindex];
+    $match = $record[$matchindex];
+    $replace = $record[$replaceindex];
+
+    // Making sure 'replace' field is not empty, in case user miss to fill them.
+    // TOOO: There may be cases whe we want to replace with empty string, probably we should add a flag for that.
+    if (empty($replace)) {
+        debugging("Skipping record with empty replace value: $table, $columnname, $id, $match", DEBUG_DEVELOPER);
```

Comment:
I think this should be normal, ie if you upload an empty file with no replacements then just do nothing. so I don't think we need debugging here, but we could report on the count of skipped rows at the end

Reply:
I have added this in the progress bar message